src/java.base/share/classes/java/lang/reflect/code/interpreter/Interpreter.java

```diff
@@ -518,6 +608,19 @@ static Object exec(MethodHandles.Lookup l, OpContext oc, Op o) {
             MethodHandle mh = opHandle(o.opName() + "_" + o.opType().returnType(), o.opType());
             Object[] values = o.operands().stream().map(oc::getValue).toArray();
             return invoke(mh, values);
+        } else if (o instanceof CoreOps.AssertOp _assert) {
+            Body testBody = _assert.bodies.get(0);
+            Boolean testResult = (Boolean) invokeBody(l,testBody,oc,List.of());
+            if (!testResult) {
+                if (_assert.bodies.size() > 1) {
+                    Body messageBlock = _assert.bodies.get(1);
+                    String message = (String) invokeBody(l, messageBlock, oc, List.of());
```

Comment:
Seeing this, I think there's a problem coming from the initial work on `assert`. The `detail` part of the assert can have any non-void type. Consider something like
```java
@CodeReflection
static void a(int a) {
   assert false : 5 + a;
}
```
the current model for that is
```
func @"a" (%0 : int)void -> {
    %1 : Var<int> = var %0 @"a";
    assert
        ()boolean -> {
            %2 : boolean = constant @"false";
            yield %2;
        }
        ()java.lang.String -> {
            %3 : int = constant @"5";
            %4 : int = var.load %1;
            %5 : int = add %3 %4;
            yield %5;
        };
    return;
};
```
currently resulting in a CCE when using the Interpreter to run it.

----

It also seems like `args` of `invokeBody` is always `List.of()` at the moment, will that change?

Reply:
> Seeing this, I think there's a problem coming from the initial work on `assert`.

Oops, my bad. I forgot the second expression of the assert statement can produce any type:

https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-14.10

And the `AssertionError` class has constructors accepting primitive arguments. 

Should be an easy iterative fix.